src/main/java/io/supertokens/inmemorydb/queries/GeneralQueries.java

```diff
@@ -306,14 +309,155 @@ public static long getUsersCount(Start start, RECIPE_ID[] includeRecipeIds)
     }
 
     public static AuthRecipeUserInfo[] getUsers(Start start, @NotNull Integer limit, @NotNull String timeJoinedOrder,
-                                                @Nullable RECIPE_ID[] includeRecipeIds, @Nullable String userId,
-                                                @Nullable Long timeJoined)
+            @Nullable RECIPE_ID[] includeRecipeIds, @Nullable String userId,
+            @Nullable Long timeJoined,
+            @Nullable DashboardSearchTags dashboardSearchTags)
             throws SQLException, StorageQueryException {
 
         // This list will be used to keep track of the result's order from the db
         List<UserInfoPaginationResultHolder> usersFromQuery;
 
-        {
+        if (dashboardSearchTags != null) {
+            ArrayList<String> queryList = new ArrayList<>();
+            {
+                StringBuilder USER_SEARCH_TAG_CONDITION = new StringBuilder();
+
+                // if recipeIds is null we are searching across all recipes
+                if (dashboardSearchTags.providers == null && dashboardSearchTags.phoneNumbers == null) {
+                    String QUERY = "SELECT  allAuthUsersTable.*" + " FROM " + getConfig(start).getUsersTable()
```

Comment:
see what the best way of doing this query is:
- sub query vs join
- filtering and then joining vs joining and then filtering